### PR TITLE
Wrap Enumerated Datatype

### PR DESCRIPTION
**Pull request by nguyenv**
Context: https://github.com/single-cell-data/TileDB-SOMA/issues/866
Depends on: [#4051](https://github.com/TileDB-Inc/TileDB/pull/4051)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 ### DO NOT USE ON CI
 
 # Target branch: Note that this should be set to the current core release, not `dev`
-TILEDB_VERSION = "2.15.4"
+TILEDB_VERSION = "pd/experiment/enums"
```
Comment (nguyenv) [inline setup.py:24]: NOTE TO SELF: Revert back to current core release.

**Comment by nguyenv** [comment]
Errors are unrelated to changes in this PR. Fix is in #1795.

**Comment by nguyenv** [comment]
On Windows we do not get to the step of installing TileDB-Py. It is erroring out when building libtiledb.

https://github.com/TileDB-Inc/TileDB-Py/actions/runs/5390955093/jobs/9787083896?pr=1790#step:13:408
```
D:\a\TileDB-Py\TileDB-Py\build\TileDB-d43f9123\tiledb\..\tiledb/sm/serialization/array_schema.h(129,43): error C3083: 'Enumeration': the symbol to the left of a '::' must be a type (compiling source file D:\a\TileDB-Py\TileDB-Py\build\TileDB-d43f9123\tiledb\sm\c_api\tiledb.cc) [D:\a\TileDB-Py\TileDB-Py\build\TileDB-d43f9123\build\tiledb\tiledb\TILEDB_CORE_OBJECTS.vcxproj] [D:\a\TileDB-Py\TileDB-Py\build\TileDB-d43f9123\build\tiledb.vcxproj
```

**Comment by johnkerl** [comment]
[sc-30316]

**Comment by nguyenv** [comment]
Errors are unrelated to this PR. There is an ongoing fix in core for dimension labels on dense arrays.

**Comment by johnkerl** [comment]
@nguyenv can you please rebase this on top of today's `dev`? Or, I have already done so locally and can push my rebase -- if that's OK with you ...

**Comment by nguyenv** [comment]
Oh just noticed this comment. Please go ahead and push your changes.

**Comment by johnkerl** [comment]
@nguyenv I for the most recent CI fail I ran `clang-format -i tiledb/query_condition.cc` on my laptop & pushed but it doesn't like that. I'm missing some settings I think. I tried reverse-engineering a run-line from the CI job but my laptop doesn't like `--clang-format-executable` flag ... 🤔

**Comment by johnkerl** [comment]
@nguyenv I for the most recent CI fail I ran `clang-format -i tiledb/query_condition.cc` on my laptop & pushed but it doesn't like that. I'm missing some settings I think. I tried reverse-engineering a run-line from the CI job but my laptop doesn't like `--clang-format-executable` flag ... 🤔

**Comment by johnkerl** [comment]
`pip install clang-format-10` to the rescue -- forget about `apt` et al. :)

**Comment by eddelbuettel** [comment]
> pip install clang-format==10.0.1 to the rescue -- forget about apt et al. :)

Apparently clang-format-14 (which is one used by core) arrived in Ubuntu 22.04 and has been available in every release since:  https://packages.ubuntu.com/search?suite=all&section=all&arch=any&keywords=clang-format-14&searchon=names

clang-format-10, if you really want it, was in 20.04: https://packages.ubuntu.com/search?keywords=clang-format-10&searchon=names&suite=all&section=all

(And I still don't understand why our script leaves it open, just calls the unversioned `clang-format` and sets us up for needless delta between version. Oh well.)

**Review by ihnorton — APPROVED** [review]

**Pull request merged**